pandas/core/internals/blocks.py

```diff
@@ -575,22 +581,36 @@ def replace(
             #  replacing it is a no-op.
             # Note: If to_replace were a list, NDFrame.replace would call
             #  replace_list instead of replace.
-            return [self] if inplace else [self.copy()]
+            if using_cow:
+                return [self.copy(deep=False)]
+            else:
+                return [self] if inplace else [self.copy()]
 
         if mask is None:
             mask = missing.mask_missing(values, to_replace)
         if not mask.any():
```

Comment:
If we pass an axis(I think``axis=0`` should work, need to be careful of 1-D values/mask though), than we should have enough info to split the block.

Reply:
Yep agree, but right now this is not that high on my list of priorities. With the ref tracking how it is right now, this only defers the copy a little bit